### ISSUE TITLE
[documentation] [tools] Decouple Flutter installation from Android Studio

### ISSUE THREAD

**Issue by iapicca** [flutter/flutter#71368]
## Use case

1.  opinions:

   - for a number of reasons [vscode](https://code.visualstudio.com/) is a **_very_** popular, if not the most popular, IDE for flutter developers
   - developers that use `vscode` rarely, if ever, use `android studio` outside the initial installation
   - many developers, especially beginners, might feel `android studio`'s installation [mandatory](https://flutter.dev/docs/get-started/install/linux#android-setup)
2. facts:
  - it is possible to use flutter without `android studio` [example](https://ksrk.medium.com/install-flutter-without-android-studio-on-window-9d3781172912)
  - `flutter doctor` [complains](https://miro.medium.com/max/700/1*LIDGbFJne4y7j2qjNaFz1Q.png) about missing `android studio`, but ignores other IDEs

## Proposal

- [documenation] provide the steps to install  `Android platform dependencies` without android studio
- [tools] treat all major IDE / editor equally
- [tools] as for this proposal https://github.com/flutter/flutter/issues/23520 a way to install Android environment via CLI flutter would be great!

**Comment by imSaharukh** [comment]
💯 agree

**Comment by Oliverq755** [comment]
I agree with this. And truly want to flutter to consider this.

**Comment by MstrZeroOne** [comment]
Completely agree.

**Comment by Captcha123** [comment]
Yes I want to use flutter but how to download it on android please tell

**Comment by shofizone** [comment]
I don't think it would be good Idea ! 
I am expecting ... Flutter should come with android studio  as package .. like android sdk ! also one click installations !

**Comment by 2shrestha22** [comment]
Yes we don't need Android Studio for Flutter.

**Comment by Captcha123** [comment]
Ok, thanks.

**Comment by dmvvilela** [comment]
What about AVD?

**Comment by Abhishek01039** [comment]
Please refer to this https://medium.com/michael-wallace/how-to-install-android-sdk-and-setup-avd-emulator-without-android-studio-aeb55c014264
@dmvvilela

**Comment by jonahwilliams** [comment]
Android Studio is currently the only 1st party supported tool for installing the android SDK. Yes, it is possible to download the sdk_manager itself and then use that to download the remaining dependencies - however this is non-trivial. If the Flutter Tool reimplemented this logic then I'd be particularly worried about updates to either the sdk_manager or the Android SDK structure breaking any time there is an update.

That is not to say I think this is a bad idea, it would just be a lot of work. Especially setting up canary tests to ensure new releases of the Android SDK do not break anything. In the short term, there are other features that being prioritized in the tool, such as support for null safety.

For those of you commenting on this issue (and others), please refrain from posting comments like "+1" or "agree". This does not add substantially to the discussion, but does have the unfortunate side effect of sending a notification to everyone subscribed to this issue.

**Comment by iapicca** [comment]
> ## Proposal
> * [documenation] provide the steps to install  `Android platform dependencies` without android studio
> * [tools] treat all major IDE / editor equally
> * [tools] a way to [install Android environment via CLI flutter](https://github.com/flutter/flutter/issues/23520) would provide a **lighter** and [simpler](https://github.com/flutter/flutter/issues/9401) installation
> * consider including this in the [installer requirements](http://flutter.dev/go/sdk-per-platform-installers)

>[...]  That is not to say I think this is a bad idea, it would just be a lot of work. Especially setting up canary tests to ensure new releases of the Android SDK do not break anything. In the short term, there are other features that being prioritized in the tool, such as support for null safety.

@jonahwilliams 
that's totally understandable and actually I agree with that,
there are plenty of improvements I rather see landing before this one.

What about the **documentation** part of the proposal? 
This seems a relatively easy fix

**Comment by jonahwilliams** [comment]
> What about the documentation part of the proposal?

That would essentially be us documenting how android studio sets up the SDK, which is subject to change just the same as if the flutter tool implemented it.

**Comment by bartekpacia** [comment]
Hi, somet time ago I wrote a blog post about setting Android SDK without Android Studio, more as a note to future self, but @huycozy suggested to share it here so I'm doing it. Hope it will be useful to whoever looks at this issue!

https://garden.pacia.tech/suckless-androidsdk-setup

**Comment by jmagman** [comment]
There's now an `android` command line tool that will allow us to install individual dependencies. https://developer.android.com/tools/agents/android-cli

To start from scratch and test this, I uninstalled Android Studio, installed `android`, and then ran `android sdk list` to see what I had installed. Note, I still had the JRE/JDK installed, more on that later:
```
$ android sdk list
Installed packages:
  build-tools/35.0.0                                               35.0.0                             Android SDK Build-Tools 35
  build-tools/36.1.0                                               36.1.0                             Android SDK Build-Tools 36.1
  cmake/3.22.1                                                     3.22.1                             CMake 3.22.1
  cmdline-tools/latest                                             unknown         ->        20.0.0   Android SDK Command-line Tools (latest)
  emulator                                                         36.2.12         ->        36.5.11  Android Emulator
  ndk/27.0.12077973                                                27.0.12077973                      NDK (Side by side) 27.0.12077973
  ndk/28.2.13676358                                                28.2.13676358                      NDK (Side by side) 28.2.13676358
  ndk/29.0.14206865                                                29.0.14206865                      NDK (Side by side) 29.0.14206865
  platform-tools                                                   36.0.0          ->        37.0.0   Android SDK Platform-Tools
  platforms/android-36                                             2.0.0                              Android SDK Platform 36
  sources/android-36                                               1.0.0                              Sources for Android 36
  system-images/android-36/google_apis_playstore/arm64-v8a         7.0.0                              Google Play ARM 64 v8a System Image
```
I ran `android sdk remove x` for every entry (ex `android sdk remove sources/android-36`) until I had an empty package list.
```
$ android sdk list
Installed packages:
  (no installed packages)
```
Then I ran `flutter build apk`. This automatically downloaded everything I needed to actually build:
```
$ android sdk list
Installed packages:
  build-tools/35.0.0           35.0.0            Android SDK Build-Tools 35
  cmake/3.22.1                 3.22.1            CMake 3.22.1
  ndk/28.2.13676358            28.2.13676358     NDK (Side by side) 28.2.13676358
  platform-tools               37.0.0            Android SDK Platform-Tools
  platforms/android-36         2.0.0             Android SDK Platform 36
```
At this point I was missing `emulator`/`system-images` to let me create and run an emulator.

```
$ android sdk install emulator
$ android emulator create
...
Successfully created device 'medium_phone' with profile 'medium_phone'.
```
Options for emulators:
```
$ android emulator create --list-profiles
large_desktop
medium_desktop
medium_phone
medium_tablet
small_desktop
small_phone
```
```
$ android emulator start medium_phone
$ flutter doctor -v
...
    • sdk gphone64 arm64 (mobile) • emulator-5554 • android-arm64  • Android 16 (API 36) (emulator)
```
And now I can `flutter run`!

At this point, `doctor` is complaining about licenses:
```
[!] Android toolchain - develop for Android devices (Android SDK version 35.0.0)
    ✗ Android license status unknown.
      Run `flutter doctor --android-licenses` to accept the SDK licenses.
      See https://flutter.dev/to/macos-android-setup for more details.
```
`flutter doctor --android-licenses` complains about `sdkmanager`
```
Android sdkmanager not found. Update to the latest Android SDK and ensure that the cmdline-tools are installed to resolve this.
```
That requires `cmdline-tools/latest`. But I wonder, if we don't have Android Studio installed, when does not agreeing to the license hit you? I've previously agreed to the licenses on this machine, so I may not have a good test of this. Let's say it's required though.
```
$ android sdk install cmdline-tools/latest
$ flutter doctor --android-licenses
Warning: This version only understands SDK XML versions up to 3 but an SDK XML file of version 4 was encountered. This can happen if you use versions of Android Studio and the command-line tools that were released at different times.
..
All SDK package licenses accepted.
```
I don't have Android Studio installed, so the XML version warning is concerning.

What I can't seem to get for free is Java/JDK. I uninstalled them, and obviously now I can't build:
```
The operation couldn’t be completed. Unable to locate a Java Runtime.
Please visit http://www.java.com for information on installing Java
```
I re-installed Android studio, which gives me Java and a JDK:
```
...
    • Java binary at: /Applications/Android Studio.app/Contents/jbr/Contents/Home/bin/java
      This is the JDK bundled with the latest Android Studio installation on this machine.
      To manually set the JDK path, use: `flutter config --jdk-dir="path/to/jdk"`.
    • Java version OpenJDK Runtime Environment (build 21.0.10+-117844308-b1163.108)
    • All Android licenses accepted.
```
But now when I `flutter run` I get that XML warning again:
```
Warning: SDK processing. This version only understands SDK XML versions up to 3 but an SDK XML file of version 4 was encountered. This can happen if you use versions of Android Studio and the command-line tools that were released at different times.
```
and the build failure: 
```
FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':app:compileReleaseJavaWithJavac'.
> Could not resolve all files for configuration ':app:androidJdkImage'.
   > Failed to transform core-for-system-modules.jar to match attributes {artifactType=_internal_android_jdk_image, org.gradle.libraryelements=jar, org.gradle.usage=java-runtime}.
      > Execution failed for JdkImageTransform: /Users/magder/Library/Android/sdk/platforms/android-36/core-for-system-modules.jar.
         > A problem occurred starting process 'command '/Applications/Android Studio.app/Contents/jbr/Contents/Home/bin/jlink''
```
